client/api/go-client/cluster.go

```diff
@@ -62,6 +62,42 @@ func (c *Client) ClusterCreate(request *api.ClusterCreateRequest) (*api.ClusterI
 	return &cluster, nil
 }
 
+func (c *Client) ClusterSetFlags(id string, request *api.ClusterSetFlagsRequest) error {
+
+	buffer, err := json.Marshal(request)
+	if err != nil {
+		return err
+	}
+
+	// Create a request
+	req, err := http.NewRequest("POST", c.host+"/clusters/"+id+"/flags",
+		bytes.NewBuffer(buffer))
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	// Set token
+	err = c.setToken(req)
+	if err != nil {
+		return err
+	}
+
+	// Send request
+	r, err := c.do(req)
+	if err != nil {
+		return err
+	}
+
+	// OBNOX: do we need to do anything here?
+	//        go async or so like in node set or device set?
+	if r.StatusCode != http.StatusOK {
+		return utils.GetErrorFromResponse(r)
+	}
+
```

Comment:
Does that mean that `ClusterDelete()` also needs `r.Body.Close()`? And so does `ClusterList()`?
Initially I thought that it would only be needed for those calls that do read a json from the body, etc, but it does not seem to be related.

What happens if we don't close the body?

I also fail to understand the relationship with the link you posted, please explain...

Reply:
Also is it correct to close the body only after the block returning in case of failed status check?

Reply:
done